README.md

```diff
@@ -7,56 +7,78 @@
+Also known as [frees-rpc], it brings the ability to combine [RPC] protocols, services, and clients in your `Freestyle` program, thanks to [gRPC].
 
 ## Installation
 
-`frees-rpc` is cross-built for Scala `2.11.x` and `2.12.x`:
+`frees-rpc` is cross-built for Scala `2.11.x` and `2.12.x`.
+
+It's divided into multiple and different artifacts, grouped by scope:
+
+* `Provided`: used from other artifacts and transitively provided.
+* `Server`: specifically for the RPC server.
+* `Client`: focused on the RPC auto-derived clients by `frees-rpc`.
+* `Test`: useful to test `frees-rpc` applications.
+
+*Artifact Name* | *Scope* | *Mandatory* | *Description*
+--- | --- | --- | ---
+`frees-rpc-server` | Server | Yes | Needed to attach RPC Services and spin-up an RPC Server.
+`frees-rpc-client-core` | Client | Yes | Mandatory to define protocols and auto-derived clients.
+`frees-rpc-client-netty` | Client | Yes* | Optional if you use `OkHttp`, required from the client perspective.
+`frees-rpc-client-okhttp` | Client | Yes* | Optional if you use `Netty`, required from the client perspective.
+`frees-rpc-config` | Server/Client | No | It provides configuration helpers using [frees-config] to load the application configuration values.
+`frees-rpc-prometheus-server` | Server | No | Scala interceptors which can be used to monitor gRPC services using Prometheus, on the _Server_ side.
+`frees-rpc-prometheus-client` | Client | No | Scala interceptors which can be used to monitor gRPC services using Prometheus, on the _Client_ side.
+`frees-rpc-prometheus-shared` | Provided | No | Common code for both the client and the server in the prometheus scope.
+`frees-rpc-dropwizard-server` | Server | No | Scala interceptors which can be used to monitor gRPC services using Dropwizard metrics, on the _Server_ side.
+`frees-rpc-dropwizard-client` | Client | No | Scala interceptors which can be used to monitor gRPC services using Dropwizard metrics, on the _Client_ side.
+`frees-rpc-interceptors` | Provided | No | Commons related to gRPC interceptors.
+`frees-rpc-testing` | Test | No | Utilities to test out `frees-rpc` applications. It provides the `grpc-testing` library as the transitive dependency.
+`frees-rpc-common` | Provided | Provided | Common things that are used throughout the project.
+`frees-rpc-internal` | Provided | Provided | Macros.
+`frees-rpc-async` | Provided | Provided | Async instances useful for interacting with the RPC services on both sides, server and the client.
+
+* `Yes*`: on the client-side, you must choose either `Netty` or `OkHttp` as the transport layer.
+
+You can install any of these dependencies in your build as follows:
 
 [comment]: # (Start Replace)
 
 ```scala
 // required for the RPC Server:
-libraryDependencies += "io.frees" %% "frees-rpc-server"        % "0.9.0"
+libraryDependencies += "io.frees" %% "frees-rpc-server"            % "0.9.0"
```

Comment:
After PR #146, should this value be `0.10.0` or that value would be changed automatically when we would do the release?

Reply:
That will be automatically changed 👍